opm/material/fluidsystems/blackoilpvt/LiveOilPvt.hpp

```diff
@@ -502,25 +492,25 @@ class LiveOilPvt
      * \brief Returns the saturation pressure of the oil phase [Pa]
      *        depending on its mass fraction of the gas component
      *
-     * \param XoG The mass fraction of the gas component in the oil phase [-]
+     * \param Rs The surface volume of gas component dissolved in what will yield one cubic meter of oil at the surface [-]
      */
     template <class Evaluation>
     Evaluation oilSaturationPressure(unsigned regionIdx,
                                      const Evaluation& temperature,
-                                     const Evaluation& XoG) const
+                                     const Evaluation& Rs) const
     {
         typedef Opm::MathToolbox<Evaluation> Toolbox;
 
         // use the saturation pressure spline to get a pretty good initial value
-        Evaluation pSat = saturationPressureSpline_[regionIdx].eval(XoG, /*extrapolate=*/true);
+        Evaluation pSat = saturationPressureSpline_[regionIdx].eval(Rs, /*extrapolate=*/true);
         Evaluation eps = pSat*1e-11;
 
         // Newton method to do the remaining work. If the initial
         // value is good, this should only take two to three
         // iterations...
         for (int i = 0; i < 20; ++i) {
-            const Evaluation& f = saturatedOilGasMassFraction(regionIdx, temperature, pSat) - XoG;
-            const Evaluation& fPrime = ((saturatedOilGasMassFraction(regionIdx, temperature, pSat + eps) - XoG) - f)/eps;
+            const Evaluation& f = gasDissolutionFactor(regionIdx, temperature, pSat) - Rs;
```

Comment:
As this is not new to the PR it may be slightly irrelevant, but:
 - why do you need to do a Newton solve here, was it just the most convenient?
 - why do you not use the AD available with Evaluation?